### PR TITLE
Update GNN  stellargraph with DGL

### PR DESCRIPTION
**Pull request by tzemicheal**
## Description

This PR replaces stellargraph reference in the training & validation scripts. This will address the issue #945  & #925

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by mdemoret-nv** [comment]
FYI, we will be targeting 23.11 for this PR

File: models/training-tuning-scripts/fraud-detection-models/README.md [hunk 1]
```diff
@@ -59,10 +54,9 @@ export DATASET=../../dataset
...
 python training.py --training-data $DATASET/training-data/fraud-detection-training-data.csv \
 --validation-data $DATASET\validation-datafraud-detection-validation-data.csv \
-         --epoch 10 \
-         --output-xgb model/xgb.pt \ 
-         --output-hinsage model/hinsage.pt \
-         --save_model True
+         --epochs 20 \
+         --model_dir models\
+         --model-type HinSAGE
+"""
```
Comment (mdemoret-nv) [inline models/training-tuning-scripts/fraud-detection-models/README.md:60]: Should these `"""` be removed?

File: models/training-tuning-scripts/fraud-detection-models/model.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
+# SPDX-FileCopyrightText: Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (mdemoret-nv) [inline models/training-tuning-scripts/fraud-detection-models/model.py:1]: We should symlink this file in the example directory. i.e. Keep this file as the real file and create a symlink to this file in the example directory.

Also add a comment about the symlink in the readme

File: models/training-tuning-scripts/fraud-detection-models/requirements.txt [hunk 3]
```diff
@@ -1,9 +1,10 @@
...
-stellargraph==1.2.1
-tensorflow==2.12.0
-xgboost==1.6.2
\ No newline at end of file
+pandas==1.5.3
+scikit_learn==1.2.2
+torch==2.0.0+cu118
+torchmetrics==0.11.4
+tqdm==4.65.0
+xgboost==1.7.1
```
Comment (mdemoret-nv) [inline models/training-tuning-scripts/fraud-detection-models/requirements.txt]: Use a conda environment file instead

File: models/validation-inference-scripts/fraud-detection-models/inference.py [hunk 4]
```diff
@@ -12,107 +12,235 @@
...
+
+    Args:
+        model (HeteroRGCN): trained HeteroRGCN model object
+        eval_loader (NodeDataLoader): evaluation dataloader
+        feature_tensors (torch.Tensor) : test feature tensor
+        target_node (str): target node encoding.
+        device (str, optional): device runtime. Defaults to 'cpu'.
+
+    Returns:
+        List: logits, index & output embedding.
```
Comment (mdemoret-nv) [inline models/validation-inference-scripts/fraud-detection-models/inference.py:149]: Wrong docstring format

File: models/validation-inference-scripts/fraud-detection-models/inference.py [hunk 5]
```diff
@@ -12,107 +12,235 @@
...
+def main(training_data, validation_data, model_dir, target_node, output_file, model_type):
+
+    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+    meta_cols = ["client_node", "merchant_node", "fraud_label", "index", "tran_id"]
+    if model_type == "RGCN":
+        gnn_model = HeteroRGCN
+    else:
+        gnn_model = HinSAGE
+
+    print(gnn_model.__name__)
```
Comment (mdemoret-nv) [inline models/validation-inference-scripts/fraud-detection-models/inference.py:218]: Extra debugging comment

File: models/validation-inference-scripts/fraud-detection-models/model.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
+# SPDX-FileCopyrightText: Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (mdemoret-nv) [inline models/validation-inference-scripts/fraud-detection-models/model.py:1]: Use a common location for this file and symlink them together

File: models/validation-inference-scripts/fraud-detection-models/requirements.txt [hunk 7]
```diff
@@ -1,12 +1,10 @@
...
+click==8.1.3
+cuml==23.4.0
+dgl==1.0.2+cu118
+numpy==1.23.5
+pandas==1.5.3
+scikit_learn==1.2.2
+torch==2.0.0+cu118
+torchmetrics==0.11.4
+tqdm==4.65.0
+xgboost==1.7.1
```
Comment (mdemoret-nv) [inline models/validation-inference-scripts/fraud-detection-models/requirements.txt:10]: Use conda environment file

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]
- Add the necessary requirements to the example environment file at `docker/conda/environments/cuda11.8_examples.yml`
- Merge the latest `branch-23.11`
- Migrate all model, testing, and input/output files to Git LFS
- Convert all docstring to use NumpyDoc format

File: models/training-tuning-scripts/fraud-detection-models/gnn-fraud-detection-training.ipynb [hunk 8]
```diff
@@ -0,0 +1,1019 @@
+{
```
Comment (gbatmaz) [inline models/training-tuning-scripts/fraud-detection-models/gnn-fraud-detection-training.ipynb:1]: <p>fraudulent transactions might be better than fraud label transactions</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/nv-morpheus/Morpheus/pull/1032/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (gbatmaz) [inline models/training-tuning-scripts/fraud-detection-models/gnn-fraud-detection-training.ipynb:1]: <p>Could you please rephrase the last sentence?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/nv-morpheus/Morpheus/pull/1032/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

**Review by gbatmaz — APPROVED** [review]
LGTM

File: docker/conda/environments/cuda11.8_examples.yml [hunk 9]
```diff
@@ -35,6 +35,10 @@ dependencies:
     - s3fs>=2023.6
     - pip
     - wrapt=1.14.1 # ver 1.15 breaks the keras model used by the gnn_fraud_detection_pipeline
+    - dgl==1.0.2+cu118
+    - torchmetrics==0.11.4
+    - tqdm==4.65.0
```
Comment (dagardner-nv) [inline docker/conda/environments/cuda11.8_examples.yml:40]: tqdm is already specified in `docker/conda/environments/cuda11.8_dev.yml`

File: docker/conda/environments/cuda11.8_examples.yml [hunk 10]
```diff
@@ -35,6 +35,10 @@ dependencies:
     - s3fs>=2023.6
     - pip
     - wrapt=1.14.1 # ver 1.15 breaks the keras model used by the gnn_fraud_detection_pipeline
+    - dgl==1.0.2+cu118
+    - torchmetrics==0.11.4
```
Comment (dagardner-nv) [inline docker/conda/environments/cuda11.8_examples.yml:39]: Is `torchmetrics` a hard requirement or a nice-to-have?
I see it being used in `models/training-tuning-scripts/fraud-detection-models/training.py` but that appears to have its own requirements file.

Comment (tzemicheal) [reply docker/conda/environments/cuda11.8_examples.yml:39]: Yes, `torchmetrics` could be removed from this yaml file. It is not used in the pipeline except for training.

File: examples/gnn_fraud_detection_pipeline/stages/graph_sage_stage.py [hunk 11]
```diff
@@ -55,21 +56,15 @@ def __init__(self,
...
                  target_node: str = "transaction"):
-        super().__init__(c)
+        super().__init__(config)
 
-        # Must import stellargraph before loading the model
-        import stellargraph.mapper  # noqa
-        import tensorflow as tf
-
-        self._keras_model = tf.keras.models.load_model(model_hinsage_file)
+        self._dgl_model, _, self.hyperparam = load_model(model_dir)
```
Comment (dagardner-nv) [inline examples/gnn_fraud_detection_pipeline/stages/graph_sage_stage.py:66]: `self.hyperparam` doesn't appear to be in use. Maybe instead:
```suggestion
        self._dgl_model, _, __ = load_model(model_dir)
```

**Comment by dagardner-nv** [comment]
Consider renaming `examples/gnn_fraud_detection_pipeline/stages` to `examples/gnn_fraud_detection_pipeline/src` since `model.py` isn't a stage.

**Review by dagardner-nv — APPROVED** [review]

File: morpheus.code-workspace [hunk 12]
```diff
@@ -560,7 +560,44 @@
...
+				"name": "Python: GNN DGL inference",
+				"type": "python",
+				"request": "launch",
+				"program": "${workspaceFolder}/examples/gnn_fraud_detection_pipeline/run.py",
+				"console": "integratedTerminal",
+				"cwd": "${workspaceFolder}/examples/gnn_fraud_detection_pipeline",
+				"justMyCode": false,
+				"args": [
+					"--input_file",
+					"validation.csv",
```
Comment (mdemoret-nv) [inline morpheus.code-workspace:574]: You can combine these into a single argument `"--input_file=validation.csv"`

File: tests/_utils/__init__.py [hunk 13]
```diff
@@ -189,6 +190,13 @@ def make_mock_response(mock_request_session: mock.MagicMock,
     return mock_response
 
 
+def remove_module(mod_to_remove: str):
```
Comment (mdemoret-nv) [inline tests/_utils/__init__.py:193]: Need a description of what this does and when to use it

File: tests/examples/gnn_fraud_detection_pipeline/conftest.py [hunk 14]
```diff
@@ -60,40 +53,58 @@ def config_fixture(config):
     yield config
 
 
+@pytest.fixture(name="manual_seed", scope="function")
+def manual_seed_fixture(manual_seed):
```
Comment (mdemoret-nv) [inline tests/examples/gnn_fraud_detection_pipeline/conftest.py:57]: Add description on what this fixture is and when to use it.

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]
- Add typing information to function signatures

**Review by mdemoret-nv — APPROVED** [review]
